### PR TITLE
:seedling: Included unit tests for CII Best practices

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Add a test file for the CII Best Practices checker
- Add tests for different badge responses and assign different scores based on badge responses
- Change CIIBestPractices function parameter and add a check for empty raw data

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: checks/evaluation/cii_best_practices.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ const (
 )
 
 // CIIBestPractices applies the score policy for the CIIBestPractices check.
-func CIIBestPractices(name string, dl checker.DetailLogger, r *checker.CIIBestPracticesData) checker.CheckResult {
+func CIIBestPractices(name string, _ checker.DetailLogger, r *checker.CIIBestPracticesData) checker.CheckResult {
```
Comment (spencerschrock) [inline checks/evaluation/cii_best_practices.go:35]: Rather than the underscore, should we be using the logger at all?
If yes, might be worth it's own PR.

File: checks/evaluation/cii_best_practices_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+package evaluation
+
+import (
+	"testing"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/clients"
+)
+
+func TestCIIBestPractices(t *testing.T) {
```
Comment (spencerschrock) [inline checks/evaluation/cii_best_practices_test.go:23]: Seems like a lot of common structure for tests here. any reason not to use a table-driven test?

something like name, CII data pointer, and expected score?

**Review by spencerschrock — APPROVED** [review]

File: checks/evaluation/cii_best_practices_test.go (hunk 2 above)
Comment (naveensrinivasan) [reply checks/evaluation/cii_best_practices_test.go:23]: I felt this was easier for this.

File: checks/evaluation/cii_best_practices.go (hunk 1 above)
Comment (naveensrinivasan) [reply checks/evaluation/cii_best_practices.go:35]: Agreed. Another PR :)
